engine/buildkit/client.go

```diff
@@ -577,6 +577,112 @@ func (c *Client) ListenHostToContainer(
 	}, nil
 }
 
+type TerminalClient struct {
+	Stdin    io.ReadCloser
+	Stdout   io.WriteCloser
+	Stderr   io.WriteCloser
+	ResizeCh chan bkgw.WinSize
+	ErrCh    chan error
+	Close    func(exitCode int) error
+}
+
+func (c *Client) OpenTerminal(
+	ctx context.Context,
+) (*TerminalClient, error) {
+	terminalClient := session.NewTerminalClient(c.Opts.MainClientCaller.Conn())
```

Comment:
@sipsma What is the equivalent now on top of your commits?

Reply:
Ah yeah, sorry we realized we needed that big PR quickly since it ended up fixing some other pre-existing issues and merged it asap, the equivalent now is `c.getMainClientCaller()`, e.g. how it's used here: https://github.com/sipsma/dagger/blob/546c900580c7558befe2c0a9caa5e9bbfb00f9c5/engine/server/socket.go#L59-L59

Reply:
🙏